lib/mfg.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Microsemi Switchtec(tm) PCIe Management Library
+ * Copyright (c) 2019, Microsemi Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/**
+ * @file
+ * @brief Switchtec core library functions for mfg operations
+ */
+
+#include "switchtec_priv.h"
+#include "switchtec/switchtec.h"
+#include "switchtec/mfg.h"
+#include "switchtec/errors.h"
+#include "switchtec/endian.h"
+#include "switchtec/mrpc.h"
+#include "switchtec/errors.h"
+#include <unistd.h>
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include "config.h"
+
+/**
+ * @brief Ping a device with a 32-bit data.
+ * @param[in]  dev	Switchtec device handle
+ * @param[in]  ping_dw	Ping data
+ * @param[out] reply_dw	Ping reply data (2's complement of ping data)
+ * @return 0 on success, error code on failure
+ */
+int switchtec_ping(struct switchtec_dev *dev, uint32_t ping_dw,
```

Comment:
It seems it's not necessary to have this function as we already have switchtec_get_device_info(), in which we should put the test of ping_dw internally, which means it should only succeed with a successful test of internally generated ping_dw.

Reply:
That sounds like a good idea!

Reply:
Updated based on Kelvin's suggestions.